src/System.Management.Automation/FormatAndOutput/common/PSStyle.cs

```diff
@@ -0,0 +1,414 @@
+            /// </summary>
+            /// <param name="red">Byte value representing red.</param>
+            /// <param name="green">Byte value representing green.</param>
+            /// <param name="blue">Byte value representing blue.</param>
+            /// <returns>String representing ANSI code for RGB value.</returns>
+            public string FromRgb(byte red, byte green, byte blue)
+            {
+                return $"\x1b[48;2;{red};{green};{blue}m";
+            }
+
+            /// <summary>
+            /// The color set as RGB as a single number.
+            /// </summary>
+            /// <param name="rgb">RGB value specified as an integer.</param>
+            /// <returns>String representing ANSI code for RGB value.</returns>
+            public string FromRgb(int rgb)
+            {
+                byte red, green, blue;
+                blue = (byte)(rgb & 0xFF);
+                rgb >>= 8;
+                green = (byte)(rgb & 0xFF);
+                rgb >>= 8;
+                red = (byte)(rgb & 0xFF);
+
+                return FromRgb(red, green, blue);
+            }
+        }
+
+        /// <summary>
+        /// Contains formatting styles for steams and objects.
+        /// </summary>
+        public class FormattingData
+        {
+            /// <summary>
+            /// Gets or sets the accent style for formatting.
+            /// </summary>
+            public string FormatAccent { get; set; } = "\x1b[32;1m";
+
+            /// <summary>
+            /// Gets or sets the accent style for errors.
+            /// </summary>
+            public string ErrorAccent { get; set; } = "\x1b[36;1m";
+
+            /// <summary>
+            /// Gets or sets the style for error messages.
+            /// </summary>
+            public string Error { get; set; } = "\x1b[31;1m";
+
+            /// <summary>
+            /// Gets or sets the style for warning messages.
+            /// </summary>
+            public string Warning { get; set; } = "\x1b[33;1m";
+
+            /// <summary>
+            /// Gets or sets the style for verbose messages.
+            /// </summary>
+            public string Verbose { get; set; } = "\x1b[33;1m";
+
+            /// <summary>
+            /// Gets or sets the style for debug messages.
+            /// </summary>
+            public string Debug { get; set; } = "\x1b[33;1m";
+        }
+
+        /// <summary>
+        /// Gets or sets the rendering mode for output.
+        /// </summary>
+        public OutputRendering OutputRendering
+        {
+            get
+            {
+                return Utils.OutputRenderingSetting;
+            }
+
+            set
+            {
+                Utils.OutputRenderingSetting = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets value to turn off all attributes.
+        /// </summary>
+        public string Reset { get; set; } = "\x1b[0m";
```

Comment:
Maybe ANSI attribute properties should be just getters too?

Reply:
makes sense, will update